scripts/abins/input/tester.py

```diff
@@ -27,59 +29,79 @@ class Tester(object):
     }
 
     @staticmethod
-    def _prepare_data(seedname):
+    def _prepare_data(seedname: str, max_displacement_kpt: Real = float("Inf")):
         """Reads reference values from ASCII files
 
-        :param seedname: Reference data will read from the file {seedname}_data.txt, except for the atomic displacements
-            which are read from files {seedname}_atomic_displacements_data_{I}.txt, where {I} are k-point indices.
-        :type seedname: str
+        Args:
+
+            seedname:
+                Reference data will read from the file {seedname}_data.txt,
+                except for the atomic displacements which are read from files
+                {seedname}_atomic_displacements_data_{I}.txt, where {I} are
+                k-point indices.
+
+            max_displacement_kpt:
+                Highest kpt index for which displacements data is read
         """
 
         with open(abins.test_helpers.find_file(seedname + "_data.txt")) as data_file:
             correct_data = json.loads(data_file.read().replace("\n", " "))
 
         num_k = len(correct_data["datasets"]["k_points_data"]["weights"])
+
         atoms = len(correct_data["datasets"]["atoms_data"])
         array = {}
-        for k in range(num_k):
-            temp = (
-                np.loadtxt(abins.test_helpers.find_file("{seedname}_atomic_displacements_data_{k}.txt".format(seedname=seedname, k=k)))
-                .view(complex)
-                .reshape(-1)
-            )
-            total_size = temp.size
-            num_freq = int(total_size / (atoms * 3))
-            array[str(k)] = temp.reshape(atoms, num_freq, 3)
 
+        for k in range(num_k):
             freq = correct_data["datasets"]["k_points_data"]["frequencies"][str(k)]
```

Comment:
am I being daft or does this just assign the values to `freq` and then just replace them with a np array version in the same object?

Reply:
Yup, it's purely for readability I think. I suppose we could have something like
```python
def to_array_inplace(x) -> None:
    x = np.asarray(x)

for k in range(num_k):
    to_array_inplace(correct_data["datasets"]["k_points_data"]["frequencies"][str(k)])
    ...
```

Reply:
I think just add a comment saying thats what you are doing is enough. I agree the original code is more readable, it just made me do a double take.

Reply:
I'm coming around to the function version actually, which communicates what it is doing explicitly with code and reduces the number of lines in the loop.